src/aggregate/aggregate_exec.c

```diff
@@ -475,8 +494,7 @@ static int buildRequest(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
 
   // Prepare the query.. this is where the context is applied.
   if ((*r)->reqflags & QEXEC_F_IS_CURSOR) {
-    RedisModuleCtx *newctx = RedisModule_GetThreadSafeContext(NULL);
-    (*r)->reqflags |= QEXEC_F_HAS_THCTX;
+    RedisModuleCtx *newctx = RedisModule_GetDetachedThreadSafeContext(ctx);
```

Comment:
It’s actually because of a comment in the documentation regarding `GetThreadSafeContext` with NULL, that for better logging you should use the detached API